docs/src/modules/spring/pages/actions-publishing-subscribing.adoc

```diff
@@ -1 +1,100 @@
+----
+include::example$spring-eventsourced-counter/src/main/java/com/example/actions/CounterJournalToTopicAction.java[tag=sub-ESE-action]
+----
+<1> subscribing to the events from the Counter.
+<2> setting the method input type to the events produced by the counter.
+<3> any return is valid. See NOTE.
+
+The events from the entity are delivered to the Action. The implementation may vary, for this simplified example you are just logging it, but it could a forward to some other component or external service.
+
+[NOTE]
+====
+The return value of the method is an `Action.Effect` with message `0`, but can be any other of type `Action.Effect<Integer>` as the return type of the method indicates. The Kalix framework needs the response to ensure that the event was successfully processed. If no exception is thrown and the method returns a value, the framework assumes that the event was successfully processed and marks it as such. This allows the next event to be sent to the subscribing method.  However, if an exception is raised and not handled, this action will not process any more events until the necessary handling is added. It will raise the same error over and over again until the application is fixed and restarted.
+====
+
+
+== Subscribing to a Value Entity
+
+You can subscribe an Action to events from an Value Entity. To receive messages from the entity, annotate a service method `@Subscribe.ValueEntiy` and specify the class of the entity. The same applies as for subscribing to an Event Sourced Entity.
+
+
+== Subscribing to a Topic
+
+To receive messages from a from Google Cloud Pub/Sub or Apache Kafka topic, annotate the service method `@Subscribe.Topic` and specify the topic name.
+
+The events from the topic are delivered to the Action and in this simplified example they are only logged.
+
+[source,java,indent=0]
+.src/main/java/com/example/actions/CounterJournalToTopicAction.java
+----
+include::example$spring-eventsourced-counter/src/main/java/com/example/actions/CounterJournalToTopicAction.java[tag=sub-topic-action]
+----
+<1> subscribing to the events from topic 'counter-events'. 
+<2> setting the method input type to the events produced by the counter.
+<3> any return is valid. See NOTE in Subscribing to an Event Sourced Entity.
+
+
+Note that by default, Kalix assumes the message in the topic was serialized as JSON and as such deserialize it into a `CounterEvent`. You can find more about serialization in Kalix here xref:spring:serialization.adoc[]
+
+== Receiving messages from an external Topic
+
+In the above example, you consumed JSON messages from a topic that you control. If you are consuming an external topic, the message format may not be under your control and may not be JSON.
+
+In such case, the Action definition should receive an array of bytes and deserialization must be handled within the method. See xref:spring:serialization.adoc[Handling Serialization] for more information on how to deal with data formats.
```

Comment:
I'm actually not sure we support this yet for the spring-sdk.. do we?

Reply:
You are right, not yet. This was an idea I talked about with Renato but is not yet implemented.

Reply:
removed